File: abbruzzese-page/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/Mappers/UserMapper.java
package abbruzzese.webpage.webpage.server.webpage.Mappers;

import org.springframework.stereotype.Component;
import abbruzzese.webpage.webpage.server.webpage.DTO.DTOUser;
import abbruzzese.webpage.webpage.server.webpage.entities.User;

@Component(value="UserMapper")
public class UserMapper {

    public User DTOtoUser(DTOUser dtouser){
        
        User user=new User();
        user.setNombre(dtouser.getNombre());
        user.setTelefono(dtouser.getTelefono());
        user.setEmail(dtouser.getEmail());
        user.setPassword(dtouser.getPassword());
        user.setMatricula(dtouser.getMatricula());
        return user;
    }

    public DTOUser UsertoDTO(User user){
        
        DTOUser dtouser=new DTOUser();
        dtouser.setNombre(user.getNombre());
        dtouser.setTelefono(user.getTelefono());
        dtouser.setEmail(user.getEmail());
        dtouser.setPassword(user.getPassword());
        dtouser.setMatricula(user.getMatricula());
        return dtouser;
    }

    public static DTOUser toDTONameEmail(User user) {
        return new DTOUser(user.getNombre(), user.getEmail());
    }
}

File: PHP_API/MAPPER/UserMapper.php
<?php

namespace App\Mapper;

use App\Entity\User;
use App\DTO\DTOUser;

class UserMapper {

    // Método para mapear de User a DTOUser
    public static function UsertoDTO(User $user) {
        return new DTOUser(
            $user->getTelefono(),
            $user->getUsername(),
            $user->getPassword(),
            $user->getEmail(),
            $user->getMatricula()
        );
    }

    // Método para mapear de DTOUser a User
    public static function DTOtoUser(DTOUser $dtoUser) {
        return new User(
            $dtoUser->getTelefono(),
            $dtoUser->getUsername(),
            $dtoUser->getPassword(),
            $dtoUser->getEmail(),
            $dtoUser->getMatricula()
        );
    }
    
}


?>